src/SFA.DAS.EAS.Portal/Application/Commands/Cohort/CohortApprovalRequestedCommand.cs

```diff
@@ -1,16 +1,72 @@
-﻿namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
+﻿using SFA.DAS.Commitments.Api.Client.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Client.Types;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
 {
-    public class CohortApprovalRequestedCommand : ICommand
-    {        
-        public long AccountId { get; private set; }
-        public long ProviderId { get; private set; }
-        public long CommitmentId { get; private set; }
+    public class CohortApprovalRequestedCommand : Command, ICommand<CohortApprovalRequestedByProvider>
```

Comment:
We should separate the concerns of the command from the handler for that command.  Is this a command or a handler?  Stick to the single responsibility principle to keep the code clean easy to test and easy to change.

Reply:
This is essentially the same point as your previous [comment](https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/pull/1646#discussion_r290274089), which I responded to in that thread.

My personal opinion is that having a single command means cleaner code, and I find it easy to test and change!